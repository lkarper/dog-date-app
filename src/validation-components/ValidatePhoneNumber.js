import React from 'react';

const ValidatePhoneNumber = (props) => {
    const { phone, setPhoneValidationError } = props;

    const regex = /^[0-9]{3}-[0-9]{3}-[0-9]{4}$/;
    const validPhone = regex.test(phone);

    if (phone && validPhone) {
        setPhoneValidationError(null);
        return <p id="phone-validation">Phone number accepted</p>; 
    } else if (phone && !validPhone) {
        setPhoneValidationError('Phone number not valid');
        return <p id="phone-validation">Phone number does not match format '123-456-7890'</p>;
    }

    setPhoneValidationError(null);
    return <p id="phone-validation">Phone number optional, but if provided must match format '123-456-7890'</p>;
}

export default ValidatePhoneNumber;