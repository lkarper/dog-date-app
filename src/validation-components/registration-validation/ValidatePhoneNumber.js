import React, { useEffect, useState } from 'react';

const ValidatePhoneNumber = (props) => {

    const { 
        phone, 
        phoneValidationError,
        setPhoneValidationError 
    } = props;

    const [phoneValidationMessage, setPhoneValidationMessage] = useState(`Phone number optional, but if provided must match format '123-456-7890'`);

    useEffect(() => {

        const regex = /^[0-9]{3}-[0-9]{3}-[0-9]{4}$/;
        const validPhone = regex.test(phone);

        if (phone && validPhone) {
            setPhoneValidationError(null);
            setPhoneValidationMessage('Phone number properly formatted.'); 
        } else if (phone && !validPhone) {
            setPhoneValidationError('Phone number not valid.');
            setPhoneValidationMessage(`Phone number does not match format '123-456-7890'.`);
        } else {
            setPhoneValidationError(null);
            setPhoneValidationMessage(`Phone number optional, but if provided must match format '123-456-7890'.`);
        }

    }, [phone, setPhoneValidationError, setPhoneValidationMessage]);

    return (
        <p 
            className={`ValidatePhoneNumber__validator ${phoneValidationError ? 'error' : ''} ${phoneValidationMessage === 'Phone number properly formatted.' ? 'valid' : ''}`}
            id="phone-validation"
        >
            {phoneValidationMessage}
        </p>
    );
}

export default ValidatePhoneNumber;
