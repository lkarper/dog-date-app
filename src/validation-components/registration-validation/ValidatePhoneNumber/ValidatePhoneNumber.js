import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

// Phone number is optional, but if provided must be validated to ensure proper formatting
const ValidatePhoneNumber = (props) => {
    const { 
        phone, 
        phoneValidationError,
        setPhoneValidationError, 
    } = props;

    const [phoneValidationMessage, setPhoneValidationMessage] = useState(`Phone number optional, but if provided must match format '123-456-7890'`);

    useEffect(() => {
        const regex = /^[0-9]{3}-[0-9]{3}-[0-9]{4}$/;
        const validPhone = regex.test(phone);

        if (phone && validPhone) {
            setPhoneValidationError('');
            setPhoneValidationMessage('Phone number properly formatted.'); 
        } else if (phone && !validPhone) {
            setPhoneValidationError('Phone number not valid.');
            setPhoneValidationMessage(`Phone number does not match format '123-456-7890'.`);
        } else {
            setPhoneValidationError('');
            setPhoneValidationMessage(`Phone number optional, but if provided must match format '123-456-7890'.`);
        }

    }, [phone, setPhoneValidationError, setPhoneValidationMessage]);

    return (
        <p 
            className={`ValidatePhoneNumber__validator ${phoneValidationError ? 'error' : ''} ${phoneValidationMessage === 'Phone number properly formatted.' ? 'valid' : ''}`}
            id="phone-validation"
        >
            {phoneValidationMessage}
        </p>
    );
}

ValidatePhoneNumber.defaultProps = {
    phone: '', 
    phoneValidationError: '',
    setPhoneValidationError: () => {}, 
};

ValidatePhoneNumber.propTypes = {
    phone: PropTypes.string.isRequired,
    phoneValidationError: PropTypes.string.isRequired,
    setPhoneValidationError: PropTypes.func.isRequired,
};

export default ValidatePhoneNumber;
