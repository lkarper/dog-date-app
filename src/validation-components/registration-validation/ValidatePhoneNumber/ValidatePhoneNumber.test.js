import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidatePhoneNumber from './ValidatePhoneNumber';

describe('ValidatePhoneNumber component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidatePhoneNumber />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidatePhoneNumber />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
