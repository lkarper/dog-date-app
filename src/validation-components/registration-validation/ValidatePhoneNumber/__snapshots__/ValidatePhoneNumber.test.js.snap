// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidatePhoneNumber component renders the UI as expected 1`] = `
<ValidatePhoneNumber
  phone=""
  phoneValidationError=""
  setPhoneValidationError={[Function]}
>
  <p
    className="ValidatePhoneNumber__validator  "
    id="phone-validation"
  >
    Phone number optional, but if provided must match format '123-456-7890'.
  </p>
</ValidatePhoneNumber>
`;
